src/examples/libpmemobj/map/data_store.c

```diff
@@ -151,24 +176,36 @@ int main(int argc, const char *argv[]) {
 	if (!map_check(mapc, D_RW(root)->map))
 		map_destroy(mapc, &D_RW(root)->map);
 
-	/* insert random items in a transaction */
```

Comment:
Do something like this:
```
insert_func insert;
ops = parse_map_type(type, &insert);
insert(...);
```
Where the insert function returned by the parsing method contains these two code paths.